Winleafs-Server/Winleafs.Server.Api/Controllers/SpotifyController.cs

```diff
@@ -0,0 +1,148 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+using System;
+using System.Threading.Tasks;
+using Winleafs.Server.Services.Exceptions;
+using Winleafs.Server.Services.Helpers;
+using Winleafs.Server.Services.Interfaces;
+
+namespace Winleafs.Server.Api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class SpotifyController : BaseApiController
+    {
+        private IUserService _userService;
+        private ISpotifyService _spotifyService;
+
+        public SpotifyController(DbContext context, IUserService userService, ISpotifyService spotifyService) : base(context)
+        {
+            _userService = userService;
+            _spotifyService = spotifyService;
+        }
+
+        /// <remarks>
+        /// Must be GET due to Spotify authorization standards
+        /// </remarks>
+        [HttpGet]
+        [Route("authorize/{applicationId}")]
+        public async Task<IActionResult> Authorize(string applicationId)
+        {
+            Guid guidOutput;
+            if (string.IsNullOrWhiteSpace(applicationId) || !Guid.TryParse(applicationId, out guidOutput))
+            {
+                return BadRequest("Error: Invalid parameters given."); //Generic error to not give too much information
+            }
+
+            //Add the user if it is his/her first time using the spotify functionality
+            await _userService.AddUserIfNotExists(applicationId);
+
+            const string scope = "playlist-read-private playlist-read-collaborative user-read-currently-playing user-read-playback-state";
+            return Redirect($"https://accounts.spotify.com/authorize/?client_id={SpotifyClientInfo.ClientID}&response_type=code&redirect_uri={SpotifyClientInfo.RedirectURI}&scope={scope}&state={applicationId}&show_dialog=false");
```

Comment:
Create a helper for this so in future API updates (that will come) its easy to edit

Reply:
What do you mean by "this" in your comment?

Reply:
The forming of the URL with the scope

Reply:
All URLs are bound to change at some point with the spotify API and service. We should prepare for that